rust/private/clippy.bzl

```diff
@@ -26,35 +24,20 @@ load(
 )
 load("//rust/private:utils.bzl", "determine_output_hash", "find_cc_toolchain", "find_toolchain")
 
-_rust_extensions = [
-    "rs",
-]
-
-def _is_rust_target(srcs):
-    return any([src.extension in _rust_extensions for src in srcs])
-
-def _rust_sources(target, rule):
-    srcs = []
-    if "srcs" in dir(rule.attr):
-        srcs += [f for src in rule.attr.srcs for f in src.files.to_list()]
-    if "hdrs" in dir(rule.attr):
-        srcs += [f for hdr in rule.attr.hdrs for f in hdr.files.to_list()]
-    return [src for src in srcs if src.extension in _rust_extensions]
-
 def _clippy_aspect_impl(target, ctx):
     if rust_common.crate_info not in target:
         return []
-    rust_srcs = _rust_sources(target, ctx.rule)
 
     toolchain = find_toolchain(ctx)
     cc_toolchain, feature_configuration = find_cc_toolchain(ctx)
     crate_info = target[rust_common.crate_info]
+    rust_srcs = crate_info.srcs.to_list()
     crate_type = crate_info.type
 
     if crate_info.is_test:
         root = crate_info.root
```

Comment:
is there a reason we wouldn't just always use the `crate_info.root` (i.e. in both the `is_test` and `not is_test` cases)?

Reply:
Yeah, that's a good call. In all use cases of `rust_common.crate_info` `root` appears to be set correctly which makes this work redundant too.
https://github.com/bazelbuild/rules_rust/blob/feaeb7ab712da45c1c94f7950e799d79e367ddeb/rust/private/rust.bzl#L252-L264
https://github.com/bazelbuild/rules_rust/blob/feaeb7ab712da45c1c94f7950e799d79e367ddeb/rust/private/rust.bzl#L287-L299
https://github.com/bazelbuild/rules_rust/blob/feaeb7ab712da45c1c94f7950e799d79e367ddeb/rust/private/rust.bzl#L404-L416
https://github.com/bazelbuild/rules_rust/blob/feaeb7ab712da45c1c94f7950e799d79e367ddeb/rust/private/rust.bzl#L419-L431
https://github.com/bazelbuild/rules_rust/blob/feaeb7ab712da45c1c94f7950e799d79e367ddeb/proto/proto.bzl#L225-L237